packages/jest-snapshot/src/State.js

```diff
@@ -83,7 +83,7 @@ class SnapshotState {
 
     const isEmpty = Object.keys(this._snapshotData).length === 0;
 
-    if ((this._dirty || this._uncheckedKeys.size) && !isEmpty) {
+    if ((this._dirty || this._uncheckedKeys.size || update) && !isEmpty) {
```

Comment:
Updated. I'm making snapshot "dirty" if it is in update mode and would throw invalid version error. I'm not very happy with returning a tuple from `getSnapshotData` but it's past midnight here and it seems to work :p